### PR TITLE
API-18170: Add index actions to fetch submissions filtered by veteran

### PR DESCRIPTION
**Pull request by hinzed1127**
## Description of change

This PR does a few things:
- Adds a background job updating new HLR, ~NOD~ (NOD can't require SSN, will require some more work to query MPI without SSN), and SC submissions with the submitting veteran's ICN on create
- Adds an index action to all HLR and SC controllers that derives a Veteran's ICN from the request headers and returns submissions filtered by their ICN

## Original issue(s)
https://vajira.max.gov/browse/API-18170

### CODE REVIEW

File: modules/appeals_api/app/controllers/appeals_api/v2/decision_reviews/higher_level_reviews_controller.rb [hunk 1]
```diff
@@ -23,9 +23,15 @@ class AppealsApi::V2::DecisionReviews::HigherLevelReviewsController < AppealsApi
   )['definitions']['hlrCreateParameters']['properties'].keys
   SCHEMA_ERROR_TYPE = Common::Exceptions::DetailedSchemaErrors
 
+  def index
+    veteran_hlrs = AppealsApi::HigherLevelReview.where(veteran_icn: target_veteran.mpi_icn)
```
Comment (cilestin) [inline modules/appeals_api/app/controllers/appeals_api/v2/decision_reviews/higher_level_reviews_controller.rb:27]: Let's order these by creation date.

File: modules/appeals_api/app/workers/appeals_api/add_icn_updater.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require 'sidekiq'
+
+module AppealsApi
+  class AddIcnUpdater
+    include Sidekiq::Worker
```
Comment (cilestin) [inline modules/appeals_api/app/workers/appeals_api/add_icn_updater.rb:7]: So by default a sidekiq worker will run for ~21 days unless you specify a different number of retries. In its current state, there's a possibility that we would be unable to get the veteran's ICN from the record were its PII to be removed (which can happen in as little as 7 days after submission). 

We should look for & gracefully exit from the job if the PII has been removed. I don't think we'd need to alert in that case, but maybe log it?

File: modules/appeals_api/app/controllers/appeals_api/v2/decision_reviews/higher_level_reviews_controller.rb [hunk 3]
```diff
@@ -23,9 +23,15 @@ class AppealsApi::V2::DecisionReviews::HigherLevelReviewsController < AppealsApi
...
+    render json: AppealsApi::HigherLevelReviewSerializer.new(veteran_hlrs).serializable_hash
+  end
+
   def create
     @higher_level_review.save
     AppealsApi::PdfSubmitJob.perform_async(@higher_level_review.id, 'AppealsApi::HigherLevelReview', 'V2')
+    AppealsApi::AddIcnUpdater.perform_async(@higher_level_review)
```
[the remark this answers is not in the document]
Comment (hinzed1127) [reply modules/appeals_api/app/controllers/appeals_api/v2/decision_reviews/higher_level_reviews_controller.rb:35]: yeah, that makes sense. I wasn't sure how to handle the fact that different classes get passed in depending on the submission type, but I see the pattern that `PdfSubmitJob` does, so I can follow that.

File: modules/appeals_api/app/workers/appeals_api/add_icn_updater.rb (hunk 2 above)
Comment (hinzed1127) [reply modules/appeals_api/app/workers/appeals_api/add_icn_updater.rb:7]: I added [this](https://github.com/department-of-veterans-affairs/vets-api/pull/10418/commits/16736b784d8e8440c5d2db72064c909ea60852f7). Is the Rails logger sufficient, or would we want to send this to Sentry and/or attach the appeal ID to the log?

Comment (cilestin) [reply modules/appeals_api/app/workers/appeals_api/add_icn_updater.rb:7]: Rails logging is fine and yes, let's add the ID into the log.

File: modules/appeals_api/config/schemas/v2/10182_headers.json [hunk 4]
```diff
@@ -12,6 +12,7 @@
     "nodCreateParameters": {
       "type": "object",
       "properties": {
+        "X-VA-SSN":             { "$ref": "#/definitions/X-VA-SSN" },
```
Comment (hinzed1127) [inline modules/appeals_api/config/schemas/v2/10182_headers.json:15]: I added SSN as a header to NOD to get index action working. It wouldn't be required outside of that index action, but would throw an error requiring needing the SSN on the index. Is this problematic anyway?

**Review by cilestin — APPROVED** [review]
LGTM!

**Pull request merged**